tests/userModel.test.js

```diff
@@ -0,0 +1,80 @@
+const mongoose = require('mongoose');
+
+const User = require('../src/models/db/user');
+const sampleUserData = require('../sample-data/sampleUserData.json');
+
+const validUserData = JSON.parse(JSON.stringify(sampleUserData));
+
+describe('User data validation', () => {
+  it('should accept valid user data', async () => {
+    const validUser = new User(validUserData);
+
+    await expect(validUser.validate()).resolves.toBeUndefined();
+  });
+
+  it.each(['email', 'password', 'userName'])(
+    'should reject user data which has no `%s` field',
+    async (field) => {
+      const incompleteUserData = {
+        ...validUserData,
+        ...{ [field]: undefined }
+      };
+      const incompleteUser = new User(incompleteUserData);
+
+      await expect(incompleteUser.validate()).rejects.toThrow(
+        mongoose.Error.ValidationError
+      );
```

Comment:
This looks a bit verbose. Let's simplify it like so:

```suggestion
     const validateIncompleteUserData = incompleteUser.validate();
     
      await expect(validateIncompleteUserData ).rejects.toThrow(
        mongoose.Error.ValidationError
      );
```

Reply:
Is it really less verbose? It's no big deal, but we now have more lines, more tokens, and a new variable

Reply:
Okay. Let's leave it as it is.